sharding/indexHashedNodesCoordinator.go

```diff
@@ -873,6 +922,53 @@ func (ihgs *indexHashedNodesCoordinator) ValidatorsWeights(validators []Validato
 	return weights, nil
 }
 
+func (ihgs *indexHashedNodesCoordinator) createActuallyLeavingPerShards(
+	unstakeLeaving map[uint32][]Validator,
+	additionalLeaving map[uint32][]Validator,
+	leaving []Validator,
+) (map[uint32][]Validator, map[uint32][]Validator) {
+	actuallyLeaving := make(map[uint32][]Validator)
+	actuallyRemaining := make(map[uint32][]Validator)
+
+	for shardId, leavingValidatorsPerShard := range unstakeLeaving {
```

Comment:
fixed it using searchInMap. I think there was a bug in the old code because if it was not found in one map it would be automatically added to the actuallyRemaining.